src/Core/Utils/Attribs.hpp

```diff
@@ -138,19 +139,36 @@ class AttribManager
 {
   public:
     using value_type = AttribBase*;
-    using Container = std::vector<value_type>;
+    using Container = Ra::Core::IndexMap<value_type>;
```

Comment:
Are we sure to switch from vector to deque as container structure ?

Reply:
When adding the tests for deep copy, i run into issues regarding deleted attributes.
The thing with the vector, we have to manage attributes deletion in a correct way, which was not the case: the test application crashed (segfault) because some `Attrib *` were null, and thus I had to check for it in a loop like
```
for ( auto attr : manager.attribs() )
```
which I think shouldn't be mandatory.

Using an `IndexMap` provides a nice and simple way to deal with attribute suppresion.
However, regarding `Handle` invalidation, the map will segfault when trying to access a deleted attribute from an old `Handle` to it.
This might be avoided by returning a `nullptr` if a `Handle` points to an invalid map entry.
Also to be noted that if the entry pointed by the old `Handle` has been replaced, this might be a problem too.

Reply:
Random access to a deque element is constant time (https://en.cppreference.com/w/cpp/container/deque)

Reply:
>This might be avoided by returning a `nullptr` if a `Handle` points to an invalid map entry.

I'm ok with this. If @dlyr agrees too, it can be added to the code.

> Also to be noted that if the entry pointed by the old Handle has been replaced, this might be a problem too.

This needs to be mentioned in the user documentation

Reply:
well, the index map access, with at(index), perform a find on the deque ... I'm not convienced here we can afford the performance penality